
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Genomics Aotearoa & NeSI">
      
      
        <link rel="canonical" href="https://genomicsaotearoa.github.io/Outlier_Analysis_Workshop/day_1/3.vcftools/">
      
      
        <link rel="prev" href="../2.pcadapt/">
      
      
        <link rel="next" href="../../day_2/4.bayescan/">
      
      <link rel="icon" href="../../theme_images/nesi_ga.png">
      <meta name="generator" content="mkdocs-1.4.3, mkdocs-material-9.1.15">
    
    
      
        <title>VCFtools - Outlier Analysis Workshop</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.26e3688c.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.ecc896b0.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
        html.glightbox-open { overflow: initial; height: 100%; }
        .gslide-title { margin-top: 0px; user-select: text; }
        .gslide-desc { color: #666; user-select: text; }
        .gslide-image img { background: white; }
        
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
            </style> <script src="../../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#vcftools-windowed-fst" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Outlier Analysis Workshop" class="md-header__button md-logo" aria-label="Outlier Analysis Workshop" data-md-component="logo">
      
  <img src="../../theme_images/nesi_ga.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Outlier Analysis Workshop
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              VCFtools
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
          
            
            
            
            <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
            
              <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_3" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"/></svg>
              </label>
            
          
            
            
            
            <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_2">
            
              <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
              </label>
            
          
            
            
            
            <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_3">
            
              <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_2" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
              </label>
            
          
        </form>
      
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsAotearoa/Outlier_Analysis_Workshop" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/Outlier_Analysis_Workshop
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  <li class="md-tabs__item">
    <a href="../.." class="md-tabs__link">
      Home
    </a>
  </li>

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="../1.intro_and_setup/" class="md-tabs__link md-tabs__link--active">
        Day 1
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../day_2/4.bayescan/" class="md-tabs__link">
        Day 2
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Outlier Analysis Workshop" class="md-nav__button md-logo" aria-label="Outlier Analysis Workshop" data-md-component="logo">
      
  <img src="../../theme_images/nesi_ga.png" alt="logo">

    </a>
    Outlier Analysis Workshop
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsAotearoa/Outlier_Analysis_Workshop" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/Outlier_Analysis_Workshop
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
      
      
      
        <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../1.intro_and_setup/" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../2.pcadapt/" class="md-nav__link">
        PCAdapt
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
      <a href="./" class="md-nav__link md-nav__link--active">
        VCFtools
      </a>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
        
      
      <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../day_2/4.bayescan/" class="md-nav__link">
        BayeScan
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../day_2/5.baypass/" class="md-nav__link">
        BayPass
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../day_2/6.conclude/" class="md-nav__link">
        Conclusion
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="vcftools-windowed-fst">VCFtools Windowed Fst<a class="headerlink" href="#vcftools-windowed-fst" title="Permanent link">&para;</a></h1>
<p>The VCFTools manual is available <a href="https://vcftools.github.io/man_latest.html">here</a>.</p>
<div class="admonition abstract">
<p class="admonition-title">Citation</p>
<p>Danecek, P., Auton, A., Abecasis, G., Albers, C. A., Banks, E., DePristo, M. A., Handsaker, R. E., Lunter, G., Marth, G. T., Sherry, S. T., McVean, G., Durbin, R., &amp; Genomes Project Analysis, G. (2011). The variant call format and VCFtools. Bioinformatics, 27(15), 2156-2158. <a href="https://doi.org/10.1093/bioinformatics/btr330">https://doi.org/10.1093/bioinformatics/btr330</a> </p>
</div>
<p>F<sub>st</sub> outliers help us to identify SNPs that behave abnormally in pairwise comparisons between populations.</p>
<p>We first need to use our metadata file (currently defined by the environmental variable <code>METADATA</code>) to make three individual files containing only the list of individuals in each population. We can do this by subsetting our sample metadata file, using the <code>grep</code> command to select lines that match each population's name, and then using <code>awk</code> to keep only the first column of metadata, i.e., the sample names.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="c1"># Load modules</span>
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>module<span class="w"> </span>purge
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>module<span class="w"> </span>load<span class="w"> </span>VCFtools/0.1.15-GCC-9.2.0-Perl-5.30.1
<a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>
<a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a><span class="c1"># Enter data directory</span>
<a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a><span class="nb">cd</span><span class="w"> </span><span class="nv">$DIR</span>/data
<a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a>
<a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a><span class="c1"># Subset metadata</span>
<a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a>grep<span class="w"> </span><span class="s2">&quot;Lemon&quot;</span><span class="w"> </span><span class="nv">$METADATA</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>awk<span class="w"> </span><span class="s1">&#39;{print $1}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span>individuals_Lemon.txt
<a id="__codelineno-0-10" name="__codelineno-0-10" href="#__codelineno-0-10"></a>grep<span class="w"> </span><span class="s2">&quot;War&quot;</span><span class="w"> </span><span class="nv">$METADATA</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>awk<span class="w"> </span><span class="s1">&#39;{print $1}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span>individuals_War.txt
<a id="__codelineno-0-11" name="__codelineno-0-11" href="#__codelineno-0-11"></a>grep<span class="w"> </span><span class="s2">&quot;Nowra&quot;</span><span class="w"> </span><span class="nv">$METADATA</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>awk<span class="w"> </span><span class="s1">&#39;{print $1}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span>individuals_Nowra.txt
</code></pre></div>
</div>
<p>Now we can pick two populations to compare. Let's work with Lemon (short for Lemon Tree, QLD, AU) and War (short for Warnambool, VIC, AU) and perform an SNP-based F<sub>st</sub> comparison.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="nb">cd</span><span class="w"> </span><span class="nv">$DIR</span>/analysis/vcftools_fst
<a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>
<a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>vcftools<span class="w"> </span>--vcf<span class="w"> </span><span class="nv">$VCF</span><span class="w"> </span>--weir-fst-pop<span class="w"> </span><span class="nv">$DIR</span>/data/individuals_Lemon.txt<span class="w"> </span>--weir-fst-pop<span class="w"> </span><span class="nv">$DIR</span>/data/individuals_War.txt<span class="w"> </span>--out<span class="w"> </span>lemon_war
<a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a>
<a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a>head<span class="w"> </span>-n<span class="w"> </span><span class="m">5</span><span class="w"> </span>lemon_war.weir.fst
</code></pre></div>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>VCFtools - 0.1.15
<a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>(C) Adam Auton and Anthony Marcketta 2009
<a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>
<a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a>Parameters as interpreted:
<a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a>        --vcf /nesi/nobackup/nesi02659/outlier_analysis_workshop/data/starling_3populations.recode.vcf
<a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a>        --weir-fst-pop /nesi/nobackup/nesi02659/outlier_analysis_workshop/data/individuals_Lemon.txt
<a id="__codelineno-2-7" name="__codelineno-2-7" href="#__codelineno-2-7"></a>        --weir-fst-pop /nesi/nobackup/nesi02659/outlier_analysis_workshop/data/individuals_War.txt
<a id="__codelineno-2-8" name="__codelineno-2-8" href="#__codelineno-2-8"></a>        --keep /nesi/nobackup/nesi02659/outlier_analysis_workshop/data/individuals_Lemon.txt
<a id="__codelineno-2-9" name="__codelineno-2-9" href="#__codelineno-2-9"></a>        --keep /nesi/nobackup/nesi02659/outlier_analysis_workshop/data/individuals_War.txt
<a id="__codelineno-2-10" name="__codelineno-2-10" href="#__codelineno-2-10"></a>        --out lemon_war
<a id="__codelineno-2-11" name="__codelineno-2-11" href="#__codelineno-2-11"></a>
<a id="__codelineno-2-12" name="__codelineno-2-12" href="#__codelineno-2-12"></a>Keeping individuals in &#39;keep&#39; list
<a id="__codelineno-2-13" name="__codelineno-2-13" href="#__codelineno-2-13"></a>After filtering, kept 26 out of 39 Individuals
<a id="__codelineno-2-14" name="__codelineno-2-14" href="#__codelineno-2-14"></a>Outputting Weir and Cockerham Fst estimates.
<a id="__codelineno-2-15" name="__codelineno-2-15" href="#__codelineno-2-15"></a>Weir and Cockerham mean Fst estimate: 0.0067306
<a id="__codelineno-2-16" name="__codelineno-2-16" href="#__codelineno-2-16"></a>Weir and Cockerham weighted Fst estimate: 0.015
<a id="__codelineno-2-17" name="__codelineno-2-17" href="#__codelineno-2-17"></a>After filtering, kept 5007 out of a possible 5007 Sites
<a id="__codelineno-2-18" name="__codelineno-2-18" href="#__codelineno-2-18"></a>Run Time = 0.00 seconds
</code></pre></div>
</details>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>CHROM   POS     WEIR_AND_COCKERHAM_FST
<a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>starling4       107735  0.160891
<a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>starling4       137462  -0.0805785
<a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a>starling4       151332  0.0524489
<a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a>starling4       227887  0.0569961
</code></pre></div>
</details>
</div>
<p>The important column is column 5: the Weighted Fst, from <a href="https://www.jstor.org/stable/2408641">Weir and Cockerham’s 1984 publication</a>.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a>wc<span class="w"> </span>-l<span class="w"> </span>lemon_war.weir.fst<span class="w"> </span>
</code></pre></div>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>5008 lemon_war.weir.fst
</code></pre></div>
</details>
</div>
<p>Notice how there are as many lines as there are SNPs in the data set, plus one for a header. It is always a good idea to check your output to ensure everything looks as expected!</p>
<p>Next, instead of calculating pairwise population differentiation on an SNP-by-SNP basis, we will use a sliding window approach. The <code>--fst-window-size 50000</code> refers to the window size of the genome (in base pairs) in which we are calculating one value: all SNPs within this window are used to calculate Fst. The <code>--fst-window-step</code> option indicates how many base pairs the window is moving down the genome before calculating F<sub>st</sub> for the second window, then the third, and so on. </p>
<div class="admonition warning">
<p class="admonition-title">Sliding windows</p>
<p>These sliding windows only work on ordered SNPs on the same chromosome/scaffold/contig. If your data is not set up like this (i.e., all your SNPs are on a single pseudo-chromosome), then this method is not appropriate for your data, as it will make an assumption about where the SNPs are located with respect to one another.</p>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>vcftools<span class="w"> </span>--vcf<span class="w"> </span><span class="nv">$VCF</span><span class="w"> </span>--fst-window-size<span class="w"> </span><span class="m">50000</span><span class="w"> </span>--fst-window-step<span class="w"> </span><span class="m">10000</span><span class="w"> </span>--weir-fst-pop<span class="w"> </span><span class="nv">$DIR</span>/data/individuals_Lemon.txt<span class="w"> </span>--weir-fst-pop<span class="w"> </span><span class="nv">$DIR</span>/data/individuals_War.txt<span class="w"> </span>--out<span class="w"> </span>lemon_war
<a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>
<a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>head<span class="w"> </span>-n<span class="w"> </span><span class="m">5</span><span class="w"> </span>lemon_war.windowed.weir.fst
</code></pre></div>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>VCFtools - 0.1.15
<a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>(C) Adam Auton and Anthony Marcketta 2009
<a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>
<a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>Parameters as interpreted:
<a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>        --vcf /nesi/nobackup/nesi02659/outlier_analysis_workshop/data/starling_3populations.recode.vcf
<a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a>        --fst-window-size 50000
<a id="__codelineno-7-7" name="__codelineno-7-7" href="#__codelineno-7-7"></a>        --fst-window-step 10000
<a id="__codelineno-7-8" name="__codelineno-7-8" href="#__codelineno-7-8"></a>        --weir-fst-pop /nesi/nobackup/nesi02659/outlier_analysis_workshop/data/individuals_Lemon.txt
<a id="__codelineno-7-9" name="__codelineno-7-9" href="#__codelineno-7-9"></a>        --weir-fst-pop /nesi/nobackup/nesi02659/outlier_analysis_workshop/data/individuals_War.txt
<a id="__codelineno-7-10" name="__codelineno-7-10" href="#__codelineno-7-10"></a>        --keep /nesi/nobackup/nesi02659/outlier_analysis_workshop/data/individuals_Lemon.txt
<a id="__codelineno-7-11" name="__codelineno-7-11" href="#__codelineno-7-11"></a>        --keep /nesi/nobackup/nesi02659/outlier_analysis_workshop/data/individuals_War.txt
<a id="__codelineno-7-12" name="__codelineno-7-12" href="#__codelineno-7-12"></a>        --out lemon_war
<a id="__codelineno-7-13" name="__codelineno-7-13" href="#__codelineno-7-13"></a>
<a id="__codelineno-7-14" name="__codelineno-7-14" href="#__codelineno-7-14"></a>Keeping individuals in &#39;keep&#39; list
<a id="__codelineno-7-15" name="__codelineno-7-15" href="#__codelineno-7-15"></a>After filtering, kept 26 out of 39 Individuals
<a id="__codelineno-7-16" name="__codelineno-7-16" href="#__codelineno-7-16"></a>Outputting Windowed Weir and Cockerham Fst estimates.
<a id="__codelineno-7-17" name="__codelineno-7-17" href="#__codelineno-7-17"></a>Weir and Cockerham mean Fst estimate: 0.0067306
<a id="__codelineno-7-18" name="__codelineno-7-18" href="#__codelineno-7-18"></a>Weir and Cockerham weighted Fst estimate: 0.015
<a id="__codelineno-7-19" name="__codelineno-7-19" href="#__codelineno-7-19"></a>After filtering, kept 5007 out of a possible 5007 Sites
<a id="__codelineno-7-20" name="__codelineno-7-20" href="#__codelineno-7-20"></a>Run Time = 1.00 seconds
</code></pre></div>
</details>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>CHROM   BIN_START       BIN_END N_VARIANTS      WEIGHTED_FST    MEAN_FST
<a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>starling4       60001   110000  1       0.160891        0.160891
<a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a>starling4       70001   120000  1       0.160891        0.160891
<a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a>starling4       80001   130000  1       0.160891        0.160891
<a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a>starling4       90001   140000  2       -0.00374291     0.0401563
</code></pre></div>
</details>
</div>
<p>Notice the output is different.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>wc<span class="w"> </span>-l<span class="w"> </span>lemon_war.windowed.weir.fst<span class="w"> </span>
</code></pre></div>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a>10838
</code></pre></div>
</details>
</div>
<p>Notice the line count is different from the SNP-based F<sub>st</sub> comparison; there are more lines in the sliding window-based Fst comparison. This is because there are more sliding windows across the chromosome in this data set than there are SNPs. Consider which of these steps is better for your data: the sliding window approach might not be the best in low-density SNP datasets.</p>
<p>Now let us plot the F<sub>st</sub> across the chromosome. To do this, we will add line numbers on our Fst file that will be used to order the F<sub>st</sub> measurements across the x-axis of our Manhattan plot.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><strong>X-axis values</strong> in the following plot are done using each outlier window's line number, as they are in order along the genome. Outlier windows are equally spaced, so line numbers are sufficient to capture the patterns along the genome. Consider that if you are plotting F<sub>st</sub> values for SNPs (rather than windows), they may not be equally spaced along the genome, so SNP position may need to be used to make your Manhattan plots.</p>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>awk<span class="w"> </span><span class="s1">&#39;{print $0&quot;\t&quot;NR}&#39;</span><span class="w"> </span>./lemon_war.windowed.weir.fst<span class="w">  </span>&gt;<span class="w"> </span>lemon_war.windowed.weir.fst.edit
<a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a>
<a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a>module<span class="w"> </span>load<span class="w"> </span>R-bundle-Bioconductor/3.15-gimkl-2022a-R-4.2.1
<a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a>R
</code></pre></div>
</div>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a><span class="nf">library</span><span class="p">(</span><span class="s">&quot;ggplot2&quot;</span><span class="p">)</span>
<a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a>
<a id="__codelineno-12-3" name="__codelineno-12-3" href="#__codelineno-12-3"></a><span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;~/outlier_analysis/analysis/vcftools_fst&quot;</span><span class="p">)</span>
<a id="__codelineno-12-4" name="__codelineno-12-4" href="#__codelineno-12-4"></a>
<a id="__codelineno-12-5" name="__codelineno-12-5" href="#__codelineno-12-5"></a><span class="n">windowed_fst</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="s">&quot;lemon_war.windowed.weir.fst.edit&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<a id="__codelineno-12-6" name="__codelineno-12-6" href="#__codelineno-12-6"></a><span class="nf">str</span><span class="p">(</span><span class="n">windowed_fst</span><span class="p">)</span>
<a id="__codelineno-12-7" name="__codelineno-12-7" href="#__codelineno-12-7"></a>
<a id="__codelineno-12-8" name="__codelineno-12-8" href="#__codelineno-12-8"></a><span class="nf">quantile</span><span class="p">(</span><span class="n">windowed_fst</span><span class="o">$</span><span class="n">WEIGHTED_FST</span><span class="p">,</span><span class="w"> </span><span class="n">probs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">.</span><span class="m">95</span><span class="p">,</span><span class="w"> </span><span class="n">.</span><span class="m">99</span><span class="p">,</span><span class="w"> </span><span class="n">.</span><span class="m">999</span><span class="p">))</span>
</code></pre></div>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a>&#39;data.frame&#39;:   10837 obs. of  7 variables:
<a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a>$ CHROM       : chr  &quot;starling4&quot; &quot;starling4&quot; &quot;starling4&quot; &quot;starling4&quot; ...
<a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a>$ BIN_START   : int  60001 70001 80001 90001 100001 110001 120001 130001 140001 150001 ...
<a id="__codelineno-13-4" name="__codelineno-13-4" href="#__codelineno-13-4"></a>$ BIN_END     : int  110000 120000 130000 140000 150000 160000 170000 180000 190000 200000 ...
<a id="__codelineno-13-5" name="__codelineno-13-5" href="#__codelineno-13-5"></a>$ N_VARIANTS  : int  1 1 1 2 2 2 2 2 1 1 ...
<a id="__codelineno-13-6" name="__codelineno-13-6" href="#__codelineno-13-6"></a>$ WEIGHTED_FST: num  0.16089 0.16089 0.16089 -0.00374 -0.00374 ...
<a id="__codelineno-13-7" name="__codelineno-13-7" href="#__codelineno-13-7"></a>$ MEAN_FST    : num  0.1609 0.1609 0.1609 0.0402 0.0402 ...
<a id="__codelineno-13-8" name="__codelineno-13-8" href="#__codelineno-13-8"></a>$ X1          : int  2 3 4 5 6 7 8 9 10 11 ...
</code></pre></div>
</details>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a>      95%       99%     99.9%
<a id="__codelineno-14-2" name="__codelineno-14-2" href="#__codelineno-14-2"></a>0.1948850 0.3501600 0.5741306
</code></pre></div>
</details>
</div>
<p>Choose the quantile threshold above which SNPs will be classified as outliers. In the code block below, we chose 99% (i.e., the top 1% of SNP windows).</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="nf">pdf</span><span class="p">(</span><span class="s">&quot;fst_starlings_windowed.pdf&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">5</span><span class="p">)</span>
<a id="__codelineno-15-2" name="__codelineno-15-2" href="#__codelineno-15-2"></a>
<a id="__codelineno-15-3" name="__codelineno-15-3" href="#__codelineno-15-3"></a><span class="nf">ggplot</span><span class="p">(</span><span class="n">windowed_fst</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">X1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">WEIGHTED_FST</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<a id="__codelineno-15-4" name="__codelineno-15-4" href="#__codelineno-15-4"></a><span class="w">  </span><span class="nf">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<a id="__codelineno-15-5" name="__codelineno-15-5" href="#__codelineno-15-5"></a><span class="w">  </span><span class="nf">geom_hline</span><span class="p">(</span><span class="n">yintercept</span><span class="o">=</span><span class="m">0.35</span><span class="p">,</span><span class="w"> </span><span class="n">linetype</span><span class="o">=</span><span class="s">&quot;dashed&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<a id="__codelineno-15-6" name="__codelineno-15-6" href="#__codelineno-15-6"></a><span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Window Number&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<a id="__codelineno-15-7" name="__codelineno-15-7" href="#__codelineno-15-7"></a><span class="w">  </span><span class="nf">theme_classic</span><span class="p">()</span>
<a id="__codelineno-15-8" name="__codelineno-15-8" href="#__codelineno-15-8"></a>
<a id="__codelineno-15-9" name="__codelineno-15-9" href="#__codelineno-15-9"></a><span class="nf">dev.off</span><span class="p">()</span>
<a id="__codelineno-15-10" name="__codelineno-15-10" href="#__codelineno-15-10"></a>
<a id="__codelineno-15-11" name="__codelineno-15-11" href="#__codelineno-15-11"></a><span class="nf">q</span><span class="p">()</span>
</code></pre></div>
</div>
<p><a class="glightbox" href="../../images/Fst_Windowed.PNG" data-type="image" data-width="100%" data-height="auto" data-desc-position="bottom"><img alt="windowed_fst" src="../../images/Fst_Windowed.PNG" width="600px" /></a></p>
<p>Finally, we will generate a list of outlier SNP IDs. We do this by selecting all SNPs located in the outlier windows.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a><span class="nb">cd</span><span class="w"> </span><span class="nv">$DIR</span>/analysis/vcftools_fst
<a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a>
<a id="__codelineno-16-3" name="__codelineno-16-3" href="#__codelineno-16-3"></a>cat<span class="w"> </span>lemon_war.windowed.weir.fst.edit<span class="w"> </span><span class="p">|</span><span class="w"> </span>awk<span class="w"> </span><span class="s1">&#39;$5&gt;0.3501600&#39;</span><span class="w"> </span>&gt;<span class="w"> </span>lemon_war.windowed.outliers
<a id="__codelineno-16-4" name="__codelineno-16-4" href="#__codelineno-16-4"></a>
<a id="__codelineno-16-5" name="__codelineno-16-5" href="#__codelineno-16-5"></a>wc<span class="w"> </span>-l<span class="w"> </span>lemon_war.windowed.outliers
</code></pre></div>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a>107 lemon_war_fst.windowed.outliers
</code></pre></div>
</details>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a>cut<span class="w"> </span>-f1-3<span class="w"> </span>lemon_war.windowed.outliers<span class="w"> </span>&gt;<span class="w"> </span>lemon_war.windowed.outliers.bed
<a id="__codelineno-18-2" name="__codelineno-18-2" href="#__codelineno-18-2"></a>
<a id="__codelineno-18-3" name="__codelineno-18-3" href="#__codelineno-18-3"></a>vcftools<span class="w"> </span>--vcf<span class="w"> </span><span class="nv">$VCF</span><span class="w"> </span>--bed<span class="w"> </span>lemon_war.windowed.outliers.bed<span class="w"> </span>--out<span class="w"> </span>fst_outliers<span class="w"> </span>--recode
<a id="__codelineno-18-4" name="__codelineno-18-4" href="#__codelineno-18-4"></a>
<a id="__codelineno-18-5" name="__codelineno-18-5" href="#__codelineno-18-5"></a>grep<span class="w"> </span>-v<span class="w"> </span><span class="s2">&quot;^#&quot;</span><span class="w"> </span>fst_outliers.recode.vcf<span class="w"> </span><span class="p">|</span><span class="w"> </span>awk<span class="w"> </span><span class="s1">&#39;{print $3}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span>vcftoolsfst_outlierSNPIDs.txt
<a id="__codelineno-18-6" name="__codelineno-18-6" href="#__codelineno-18-6"></a>
<a id="__codelineno-18-7" name="__codelineno-18-7" href="#__codelineno-18-7"></a>wc<span class="w"> </span>-l<span class="w"> </span>vcftoolsfst_outlierSNPIDs.txt
</code></pre></div>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a>VCFtools - 0.1.15
<a id="__codelineno-19-2" name="__codelineno-19-2" href="#__codelineno-19-2"></a>(C) Adam Auton and Anthony Marcketta 2009
<a id="__codelineno-19-3" name="__codelineno-19-3" href="#__codelineno-19-3"></a>
<a id="__codelineno-19-4" name="__codelineno-19-4" href="#__codelineno-19-4"></a>Parameters as interpreted:
<a id="__codelineno-19-5" name="__codelineno-19-5" href="#__codelineno-19-5"></a>        --vcf /nesi/nobackup/nesi02659/outlier_analysis_workshop/data/starling_3populations.recode.vcf
<a id="__codelineno-19-6" name="__codelineno-19-6" href="#__codelineno-19-6"></a>        --out fst_outliers
<a id="__codelineno-19-7" name="__codelineno-19-7" href="#__codelineno-19-7"></a>        --recode
<a id="__codelineno-19-8" name="__codelineno-19-8" href="#__codelineno-19-8"></a>        --bed lemon_war.windowed.outliers.bed
<a id="__codelineno-19-9" name="__codelineno-19-9" href="#__codelineno-19-9"></a>
<a id="__codelineno-19-10" name="__codelineno-19-10" href="#__codelineno-19-10"></a>After filtering, kept 39 out of 39 Individuals
<a id="__codelineno-19-11" name="__codelineno-19-11" href="#__codelineno-19-11"></a>Outputting VCF file...
<a id="__codelineno-19-12" name="__codelineno-19-12" href="#__codelineno-19-12"></a>        Read 107 BED file entries.
<a id="__codelineno-19-13" name="__codelineno-19-13" href="#__codelineno-19-13"></a>After filtering, kept 61 out of a possible 5007 Sites
<a id="__codelineno-19-14" name="__codelineno-19-14" href="#__codelineno-19-14"></a>Run Time = 0.00 seconds
</code></pre></div>
</details>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a>61 vcftoolsfst_outlierSNPIDs.txt
</code></pre></div>
</details>
</div>
<p>We have a total of 61 outlier SNPs located across 107 outlier SNP windows.</p>





                
              </article>
            </div>
          
          
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky"], "search": "../../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.b4d07000.min.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  <script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>