
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa &amp; NeSI" name="author"/>
<link href="https://genomicsaotearoa.github.io/Outlier_Analysis_Workshop/day_1/1.intro_and_setup/" rel="canonical"/>
<link href="../../theme_images/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.2, mkdocs-material-8.5.10" name="generator"/>
<title>A brief introduction to Genetic Outlier and Association Analysis - Outlier Analysis Workshop</title>
<link href="../../assets/stylesheets/main.472b142f.min.css" rel="stylesheet"/>
<link href="../../assets/stylesheets/palette.08040f6c.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="../../stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="../../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="" data-md-color-primary="indigo" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#a-brief-introduction-to-genetic-outlier-and-association-analysis">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--lifted" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Outlier Analysis Workshop" class="md-header__button md-logo" data-md-component="logo" href="../.." title="Outlier Analysis Workshop">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Outlier Analysis Workshop
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              A brief introduction to Genetic Outlier and Association Analysis
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/Outlier_Analysis_Workshop" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/Outlier_Analysis_Workshop
  </div>
</a>
</div>
</nav>
<nav aria-label="Tabs" class="md-tabs" data-md-component="tabs">
<div class="md-tabs__inner md-grid">
<ul class="md-tabs__list">
<li class="md-tabs__item">
<a class="md-tabs__link" href="../..">
      Home
    </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link md-tabs__link--active" href="./">
        Day 1
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day_2/4.bayescan/">
        Day 2
      </a>
</li>
</ul>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Outlier Analysis Workshop" class="md-nav__button md-logo" data-md-component="logo" href="../.." title="Outlier Analysis Workshop">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
    Outlier Analysis Workshop
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/Outlier_Analysis_Workshop" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/Outlier_Analysis_Workshop
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../..">
        Home
      </a>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" id="__nav_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2">
          Day 1
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-label="Day 1" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" data-md-toggle="toc" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          A brief introduction to Genetic Outlier and Association Analysis
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
        A brief introduction to Genetic Outlier and Association Analysis
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#define-your-working-directory-for-this-project-and-the-vcf-file-location">
    Define your working directory for this project and the VCF file location:
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#project-data">
    Project data
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../2.pcadapt/">
        PCAdapt
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../3.vcftools/">
        VCFtools Windowed Fst
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_3" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3">
          Day 2
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-label="Day 2" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day_2/4.bayescan/">
        4.bayescan
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day_2/5.baypass/">
        5.baypass
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day_2/6.conclude/">
        6.conclude
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#define-your-working-directory-for-this-project-and-the-vcf-file-location">
    Define your working directory for this project and the VCF file location:
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#project-data">
    Project data
  </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<a class="md-content__button md-icon" href="https://github.com/GenomicsAotearoa/Outlier_Analysis_Workshop/edit/master/docs/day_1/1.intro_and_setup.md" title="Edit this page">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"></path></svg>
</a>
<h1 id="a-brief-introduction-to-genetic-outlier-and-association-analysis">A brief introduction to Genetic Outlier and Association Analysis<a class="headerlink" href="#a-brief-introduction-to-genetic-outlier-and-association-analysis" title="Permanent link">Â¶</a></h1>
<p>When we look through a genome to try find loci that are under divergent selection, we often conduct what is called outlier or association analyses. <strong>Outlier analysis</strong> requires just knowledge of the genetics of your samples (plus sample metadata, for example population groupings), and tries to find loci that behave very differently from the underlying patterns across the genome (with the assumption being that the rest of the genome represents patterns of neutral genetic diveristy)<sup id="fnref:A"><a class="footnote-ref" href="#fn:A">1</a></sup>. Meanwhile <strong>association analysis</strong> require some sort of covariate data, and tests whether there are any genetic variants statistically associated with this new data (you may have heard the term <a href="https://www.genome.gov/genetics-glossary/Genome-Wide-Association-Studies">GWAS</a>). This covariate data can come in the form of phenotype data (e.g. morphology, disease status, physiology measures), or could be spatial (e.g. environmental, climate). Association tests look for sites in the genome where the presence or absence of a variant is highly correlated with the values in the co-variate data, usually through some regression type analysis.</p>
<p>Throughout this vignette I will refer to outlier and association analysis collectively as selection analysis. There are a lot of programs that exist currently. You can find a very long (though not exhaustive) list of them <a href="https://bioinformaticshome.com/tools/gwas/gwas.html">here</a>. </p>
<p>This vignette still start by covering some very simple outlier analyses:</p>
<ul>
<li><a href="https://bcm-uga.github.io/pcadapt/articles/pcadapt.html"><strong>PCAdapt</strong></a>, a program that can depect genetic marker outliers without having population<sup id="fnref:B"><a class="footnote-ref" href="#fn:B">2</a></sup> designations, using a Principle Component Analysis (PCA) approach.</li>
<li><strong>F<sub>ST</sub></strong> outlier analysis, an approach that uses pairwise comparisons between two populations and the fixation index metric to assess each genetic marker.</li>
</ul>
<!-- Prior was linked to GBIF in main/. What is GBIF? -->
<p>Next we will conduct some more advanced outlier analysis:</p>
<ul>
<li><a href="https://bcm-uga.github.io/pcadapt/articles/pcadapt.html"><strong>Bayescan</strong></a> a program.</li>
<li><a href="http://www1.montpellier.inra.fr/CBGP/software/baypass/"><strong>Baypass</strong></a>, a program that elaborates on the bayenv model (another popular association analysis program).</li>
</ul>
<p>We'll cover the pre-processing of program specific input files, how to run the programs, how to visualise the output and also in some cases we'll need to take extra steps to map the genetic markers of interest back to the SNP data.</p>
<div class="admonition note">
<p class="admonition-title"><img alt="ðŸ”°" class="twemoji" src="https://twemoji.maxcdn.com/v/latest/svg/1f530.svg" title=":beginner:"> Reduced representation verses whole genome sequencing</img></p>
<p>Completing outlier analysis is possible and often done on reduced representation data. It is important to remember how your genome coverage (the number of genome variant sites / the genome length)<sup id="fnref2:A"><a class="footnote-ref" href="#fn:A">1</a></sup> will affect your results and interpretation. Often with WGS data, you will see well resolved 'peaks' with a fairly smooth curve of points leading up to it either side. From this we often infer that the highest point is the genetic variant of interest, and the other sites either side of that exhibit signals of selection because they reside close to, and thus are linked, to the variant of interest. However, consider that even in WGS data, unless we have every single genetic variant represented (which may not be the case, depending on our variant calling and filtering parameters) it is possible that the genetic variant of interest that we have identified is not the main one, but is simply another neighbouring linked SNP to one that is not represented in the data. This problem becomes even more relevant with reduced representation sequencing (RRS), for which the genome coverage may be extremely patchy<sup id="fnref:C"><a class="footnote-ref" href="#fn:C">3</a></sup>. Thus with all outlier analysis, but especially so for those using RRS data, remember that your flagged outliers are not exhaustive, and may themselves only be liked to the variant that is truely under selection.</p>
</div>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="https://els-jbs-prod-cdn.jbs.elsevierhealth.com/cms/asset/4be56b5b-8593-4116-ab9a-ec7b9e3c9a05/gr1.jpg"><img alt="ScreenShot" src="https://els-jbs-prod-cdn.jbs.elsevierhealth.com/cms/asset/4be56b5b-8593-4116-ab9a-ec7b9e3c9a05/gr1.jpg"/></a></p>
<h2 id="define-your-working-directory-for-this-project-and-the-vcf-file-location">Define your working directory for this project and the VCF file location:<a class="headerlink" href="#define-your-working-directory-for-this-project-and-the-vcf-file-location" title="Permanent link">Â¶</a></h2>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a>mkdir /nesi/nobackup/uoa02613/kstuart_projects/outlier_analysis
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a><span class="nv">DIR</span><span class="o">=</span>/nesi/nobackup/uoa02613/kstuart_projects/outlier_analysis
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a><span class="nb">cd</span> <span class="nv">$DIR</span>
</code></pre></div>
</div>
<p>Our data tree will look like:</p>
<div class="admonition note">
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a>outlier_analysis/
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a>â”œâ”€â”€ analysis
<a href="#__codelineno-1-3" id="__codelineno-1-3" name="__codelineno-1-3"></a>â”‚   â”œâ”€â”€ bayescan
<a href="#__codelineno-1-4" id="__codelineno-1-4" name="__codelineno-1-4"></a>â”‚   â”œâ”€â”€ baypass
<a href="#__codelineno-1-5" id="__codelineno-1-5" name="__codelineno-1-5"></a>â”‚   â”œâ”€â”€ pcadapt
<a href="#__codelineno-1-6" id="__codelineno-1-6" name="__codelineno-1-6"></a>â”‚   â”œâ”€â”€ summary
<a href="#__codelineno-1-7" id="__codelineno-1-7" name="__codelineno-1-7"></a>â”‚   â””â”€â”€ vcftools_fst
<a href="#__codelineno-1-8" id="__codelineno-1-8" name="__codelineno-1-8"></a>â”œâ”€â”€ data
<a href="#__codelineno-1-9" id="__codelineno-1-9" name="__codelineno-1-9"></a>â”œâ”€â”€ programs
<a href="#__codelineno-1-10" id="__codelineno-1-10" name="__codelineno-1-10"></a>â””â”€â”€ workshop_material
</code></pre></div>
</div>
<p>So lets set up our directories to match this</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>mkdir -p <span class="o">{</span>analysis/<span class="o">{</span>bayescan,baypass,pcadapt,summary,vcftools_fst<span class="o">}</span>,data,programs,workshop_material<span class="o">}</span>
</code></pre></div>
</div>
<h2 id="project-data">Project data<a class="headerlink" href="#project-data" title="Permanent link">Â¶</a></h2>
<p>The data provided in this workshop contains 5007 SNPs loci for across 39 individuals (13 individuals each from 3 different locations). There is some missingness (i.e. missing SNP calls) within this data.</p>
<p>There is also a metadata file, that contains the individuals unique IDs, their assigned populations, and a wingspan measurement for each individual. </p>
<p>Let's grab this data from the project's git resository, place the data files into our <code>data</code> directory, and define the environmental variables <code>VCF</code> and <code>METADATA</code> with the locations of the genetic variant and metadata files respectively.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a><span class="nb">cd</span> <span class="nv">$DIR</span>/workshop_material
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a>git clone https://github.com/katarinastuart/Ev1_SelectionMetaAnalysis.git
<a href="#__codelineno-3-3" id="__codelineno-3-3" name="__codelineno-3-3"></a>cp <span class="nv">$DIR</span>/workshop_material/Ev1_SelectionMetaAnalysis/workshop_files/* <span class="nv">$DIR</span>/data
<a href="#__codelineno-3-4" id="__codelineno-3-4" name="__codelineno-3-4"></a><span class="nv">VCF</span><span class="o">=</span><span class="nv">$DIR</span>/data/starling_3populations.recode.vcf
<a href="#__codelineno-3-5" id="__codelineno-3-5" name="__codelineno-3-5"></a><span class="nv">METADATA</span><span class="o">=</span><span class="nv">$DIR</span>/data/starling_3populations_metadata.txt
</code></pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Working with your own data</p>
<p>Alternatively, you can also use your own data for this workshop. If so, it is a good idea to thin your SNP dataset down to roughly 5,000 SNPs to ensure compute times are not too long. If you have more than 50 individuals you may also want to reduce this too. If you would like to do this, just place your genetic variant and metadata file in the <code>data</code> directory and define <code>VCF</code> and <code>METADATA</code> based on their names.</p>
</div>
<p>Across this workshop, we will need the genetic data to be in several different formats. Let's prepare that now. First we convert the VCF to PLINK, and then to BED.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a><span class="nb">cd</span> <span class="nv">$DIR</span>/data
<a href="#__codelineno-4-2" id="__codelineno-4-2" name="__codelineno-4-2"></a>module load VCFtools/0.1.15-GCC-9.2.0-Perl-5.30.1
<a href="#__codelineno-4-3" id="__codelineno-4-3" name="__codelineno-4-3"></a>module load PLINK/1.09b6.16 
<a href="#__codelineno-4-4" id="__codelineno-4-4" name="__codelineno-4-4"></a>vcftools --vcf <span class="nv">$VCF</span> --plink --out starling_3populations.plink
<a href="#__codelineno-4-5" id="__codelineno-4-5" name="__codelineno-4-5"></a>plink --file starling_3populations.plink --make-bed --noweb --out starling_3populations
</code></pre></div>
</div>
<div class="footnote">
<hr/>
<ol>
<li id="fn:A">
<p>It is very important then to account for any population substructure. There are many different ways to approach this: refer to introduction slides for some guidance.Â <a class="footnote-backref" href="#fnref:A" title="Jump back to footnote 1 in the text">â†©</a><a class="footnote-backref" href="#fnref2:A" title="Jump back to footnote 1 in the text">â†©</a></p>
</li>
<li id="fn:B">
<p>I will say population for simplicity throughout this vignette. However, equally we can test for differences between sample sites, subpopulations, and other types of groupings. What counts as one 'group' of organisms will be dependent on your study system or study question.Â <a class="footnote-backref" href="#fnref:B" title="Jump back to footnote 2 in the text">â†©</a></p>
</li>
<li id="fn:C">
<p>You may also want to consider linkage blocks.Â <a class="footnote-backref" href="#fnref:C" title="Jump back to footnote 3 in the text">â†©</a></p>
</li>
</ol>
</div>
</article>
</div>
</div>
<a class="md-top md-icon" data-md-component="top" hidden="" href="#">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </a>
</main>
<footer class="md-footer">
<nav aria-label="Footer" class="md-footer__inner md-grid">
<a aria-label="Previous: Home" class="md-footer__link md-footer__link--prev" href="../.." rel="prev">
<div class="md-footer__button md-icon">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</div>
<div class="md-footer__title">
<div class="md-ellipsis">
<span class="md-footer__direction">
                Previous
              </span>
              Home
            </div>
</div>
</a>
<a aria-label="Next: PCAdapt" class="md-footer__link md-footer__link--next" href="../2.pcadapt/" rel="next">
<div class="md-footer__title">
<div class="md-ellipsis">
<span class="md-footer__direction">
                Next
              </span>
              PCAdapt
            </div>
</div>
<div class="md-footer__button md-icon">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"></path></svg>
</div>
</a>
</nav>
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky"], "search": "../../assets/javascripts/workers/search.16e2a7d4.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
<script src="../../assets/javascripts/bundle.d6c3db9e.min.js"></script>
<script src="../../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>