# Bayescan

The VCFTools manual is available [here](https://github.com/mfoll/BayeScan).

!!! abstract "Citation"

    Foll M & Gaggiotti O 2008 A Genome-Scan Method to Identify Selected Loci Appropriate for Both Dominant and Codominant Markers: A Bayesian Perspective. Genetics 180 977â€“993. (doi:10.1534/genetics.108.092221)

Bayescan identified outlier SNPs based on allele frequencies. More explanation about alpha and such.

First, we will need to convert out VCF to the Bayescan format. To do this we will use the genetic file conversion program called [PGDspider](http://www.cmpg.unibe.ch/software/PGDSpider/). 

!!! terminal "code"

    ```bash
    cd $DIR/programs
    wget http://www.cmpg.unibe.ch/software/PGDSpider/PGDSpider_2.1.1.5.zip
    unzip *.zip
    ```

We also need to create a new populations metadata file, which contains individual names in column 1, and population names in column 2.

!!! terminal "code"

    ```bash
    cd $DIR/data
    cut -f1,2 $METADATA > starling_3populations_metadata_INDPOP.txt
    ```

Now navigate to the folder in which we will be running the Bayescan analysis.

!!! terminal "code"

    ```bash
    cd $DIR/analysis/bayescan
    ```

We now run PGDSpider in two steps: 

1. Convert the VCF file to the PGD format 
1. Convert PGD format to Bayescan format. 

To do this we will need to create a SPID file called `VCF_PGD.spid` using the `nano` command. Paste in the below, replacing the location of your metadata file.

include snapshot of SPID:

!!! terminal "code"

    ```bash
    nano VCF_PGD.spid
    ```

!!! note "Write the following into `VCF_PGD.spid`"

    ```
    # VCF Parser questions 
    PARSER_FORMAT=VCF 

    # Only output SNPs with a phred-scaled quality of at least: 
    VCF_PARSER_QUAL_QUESTION= 

    # Select population definition file: 
    VCF_PARSER_POP_FILE_QUESTION=/nesi/nobackup/nesi02659/outlier_analysis_workshop/data/starling_3populations_metadata_INDPOP.txt

    # What is the ploidy of the data? 
    VCF_PARSER_PLOIDY_QUESTION=DIPLOID 

    # Do you want to include a file with population definitions? 
    VCF_PARSER_POP_QUESTION=true 

    # Output genotypes as missing if the Phred-scale genotype quality is below: 
    VCF_PARSER_GTQUAL_QUESTION= 

    # Do you want to include non-polymorphic SNPs? 
    VCF_PARSER_MONOMORPHIC_QUESTION=false 

    # Only output following individuals (ind1, ind2, ind4, ...): 
    VCF_PARSER_IND_QUESTION= 

    # Only input following regions (refSeqName:start:end, multiple regions: whitespace separated): 
    VCF_PARSER_REGION_QUESTION= 

    # Output genotypes as missing if the read depth of a position for the sample is below: 
    VCF_PARSER_READ_QUESTION= 

    # Take most likely genotype if "PL" or "GL" is given in the genotype field? 
    VCF_PARSER_PL_QUESTION=false 

    # Do you want to exclude loci with only missing data? 
    VCF_PARSER_EXC_MISSING_LOCI_QUESTION=false

    # PGD Writer questions 
    WRITER_FORMAT=PGD 
    ```

??? tip "Writing out files with `nano`"

    Once you have copied and pasted the above, press <kbd>Ctrl</kbd> + <kbd>O</kbd> then press <kbd>Enter</kbd> to save your file. Then, exit the program using <kbd>Ctrl</kbd> + <kbd>X</kbd>

Now run the two step conversion.

!!! terminal "code" 

    ```bash
    module purge
    module load Java/1.8.0_144

    java -Xmx1024m -Xms512m -jar $DIR/programs/PGDSpider_2.1.1.5/PGDSpider2-cli.jar -inputfile $VCF -inputformat VCF -outputfile starling_3populations.pgd -outputformat PGD -spid VCF_PGD.spid

    java -Xmx1024m -Xms512m -jar $DIR/programs/PGDSpider_2.1.1.5/PGDSpider2-cli.jar -inputfile starling_3populations.pgd -inputformat PGD -outputfile starling_3populations.bs -outputformat GESTE_BAYE_SCAN
    ```

Let's have a quick look at what the input file looks like:

!!! terminal "code"

    ```bash
    head starling_3populations.bs
    ```

    ??? success "Output"

        ```
        [loci]=5007

        [populations]=3

        [pop]=1
        1      12      2       9 3 
        2      20      2       11 9
        3      18      2       15 3
        4      20      2       0 20
        5      22      2       2 20
        ```

So for each population we have a note of how many REF and ALT alleles we have at each genomic variant position.

!!! note "An important note about additive genetic variance"

    It is important to bear in mind how the input genetic data for outlier or association models is being interpreted by the model. When dealing with many of these models (and input genotype files) the assumption is that the SNP effects are [additive](https://link.springer.com/referenceworkentry/10.1007/978-3-319-47829-6_5-1). This can be seen from, for example, the way we encode homozygous reference allele, heterozygous, and homozygous alternate allele as "0", "1", and "2" respectively in a BayPass input genofile. For the diploid organism (with two variant copies for each allele) one copy of a variant (i.e. heterozygous) is assumed to have half the effect of having two copies. However, what if the locus in question has dominance effects? This would mean the heterozygous form behaves the same as the homozygous dominant form, and it would be more appropriate to label these instead as "0", "0", "1". But with thousands, if not millions of (most likely) completely unknown variants in a dataset, how can we possibly know? The answer is we cannot. And most models will assume additive effects, because this the simplest assumption. However, by not factoring in dominance effects we could possible be missing many important functional variants, as [Reynolds et al. (2021)](https://www.nature.com/articles/s41588-021-00872-5) demonstrates. Genomics is full of caveats and pitfalls, which while providing new directions to explore can be a bit overwhelming. Remember, your selection analysis doesn't have to be exhaustive, just make sure it is fit for purpose within your study design. There is so much going on in just one genome, there is no way you can analyse everything in one go. 

Now let's set Bayescan to run. Using the `nano bayescan_starling.sl` we will create a Slurm script and submit it to run using the command `sbatch bayescan_starling.sl`. This should take approximately 1 hr to run. Currently everything is set to default, but read the manual if you want to understand what they mean and how to refine them if needed.

!!! terminal "code"

    ```bash
    #!/bin/bash -e
    #SBATCH --account=uoa02613
    #SBATCH --job-name=bayescan_starling.sl
    #SBATCH --time=12:00:00
    #SBATCH --mem=5GB
    #SBATCH --output=%x_%j.out
    #SBATCH --error=%x_%j.err
    #SBATCH --ntasks=1
    #SBATCH --cpus-per-task=8
    #SBATCH --profile task

    cd /nesi/nobackup/uoa02613/kstuart_projects/outlier_analysis/analysis/bayescan

    # Load bayescan
    module purge
    module load BayeScan/2.1-GCCcore-7.4.0

    # Run bayescan. 
    bayescan_2.1 ./starling_3populations.bs -od ./ -threads 8 -n 5000 -thin 10 -nbp 20 -pilot 5000 -burn 50000 -pr_odds 10
    ```
 
Identify outliers:

!!! terminal "code"

    ```bash
    module load R/4.1.0-gimkl-2020a
    R
    ```

!!! r-project "code"

    ```r
    library(ggplot2)
    setwd("/nesi/nobackup/uoa02613/kstuart_projects/outlier_analysis/analysis/bayescan")
    source("/opt/nesi/CS400_centos7_bdw/BayeScan/2.1-GCCcore-7.4.0/R\ functions/plot_R.r")
    outliers.bayescan=plot_bayescan("starling_3population_fst.txt",FDR=0.05)
    outliers.bayescan
    write.table(outliers.bayescan, file="bayescan_outliers.txt")

    q()
    ```

    ??? success "Output"
    
        ```
        $outliers
        [1] 333  395 1367 2376 3789
        $nb_outliers
        [1] 5
        ```

Mapping Outliers

!!! terminal "code"

    ```bash
    cd $DIR/analysis/bayescan
    ```

Create list of SNPs in VCF, assign line numbers that can be used to find matching line numbers in outliers (SNP ID is lost in bayescan, line numbers used as signifiers).

!!! terminal "code"

    ```bash
    grep -v "^#" ../../data/starling_3populations.recode.vcf  | cut -f1-3 | awk '{print $0"\t"NR}' > starling_3populations_SNPs.txt

    awk '{print $2}' bayescan_outliers.txt > bayescan_outliers_numbers.txt
    ```

list of outlier SNPS, by matching column 1 of of the outliers list to the fourth column of the whole SNP list data.

!!! terminal "code"

    ```bash
    awk 'FNR==NR{a[$1];next} (($4) in a)' bayescan_outliers_numbers.txt starling_3populations_SNPs.txt   | cut -f3 > bayescan_outlierSNPIDs.txt
    ```

Bayescan Log Plot, colouring the outliers in a different colour.

!!! terminal "code"

    ```bash
    module load R/4.1.0-gimkl-2020a
    R
    ```

!!! r-project "code"

    ```r
    library(ggplot2)
    library(dplyr)
    setwd("/nesi/nobackup/uoa02613/kstuart_projects/outlier_analysis/analysis/bayescan")

    bayescan.out<- read.table("starling_3population_fst.txt", header=TRUE)
    bayescan.out <- bayescan.out %>% mutate(ID = row_number())
    bayescan.outiers<- read.table("bayescan_outliers_numbers.txt", header=FALSE)
    outliers.plot <- filter(bayescan.out, ID %in% bayescan.outiers[["V1"]])

    png("bayescan_outliers.png", width=600, height=350)
    ggplot(bayescan.out, aes(x = log10.PO., y = alpha)) +
      geom_vline(
        xintercept = 0,
        linetype = "dashed",
        color = "black",
        size = 3
      ) +
      geom_point(
        aes(x = log10.PO., y = alpha),
        data = outliers.plot,
        col = "red",
        fill = "red",
        size = 5,
        alpha = 1
      ) +
      scale_x_continuous(limits = c(-1.3, 3.5)) +
      theme_classic(base_size = 18) +
      theme(axis.text = element_text(size = 18),
            axis.title = element_text(size = 22, face = "bold"))

    dev.off()

    q()
    ```

![windowed_fst](../images/bayescan_outliers.png){width="300px"}

