
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Genomics Aotearoa & NeSI">
      
      
        <link rel="canonical" href="https://genomicsaotearoa.github.io/Outlier_Analysis_Workshop/day_1/1.intro_and_setup/">
      
      
        <link rel="prev" href="../..">
      
      
        <link rel="next" href="../2.pcadapt/">
      
      <link rel="icon" href="../../theme_images/nesi_ga.png">
      <meta name="generator" content="mkdocs-1.4.3, mkdocs-material-9.1.15">
    
    
      
        <title>Introduction - Outlier Analysis Workshop</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.26e3688c.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.ecc896b0.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
        html.glightbox-open { overflow: initial; height: 100%; }
        .gslide-title { margin-top: 0px; user-select: text; }
        .gslide-desc { color: #666; user-select: text; }
        .gslide-image img { background: white; }
        
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
            </style> <script src="../../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#a-brief-introduction-to-genetic-outlier-and-association-analysis" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Outlier Analysis Workshop" class="md-header__button md-logo" aria-label="Outlier Analysis Workshop" data-md-component="logo">
      
  <img src="../../theme_images/nesi_ga.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Outlier Analysis Workshop
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Introduction
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
          
            
            
            
            <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
            
              <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_3" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"/></svg>
              </label>
            
          
            
            
            
            <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_2">
            
              <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
              </label>
            
          
            
            
            
            <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_3">
            
              <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_2" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
              </label>
            
          
        </form>
      
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsAotearoa/Outlier_Analysis_Workshop" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/Outlier_Analysis_Workshop
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  <li class="md-tabs__item">
    <a href="../.." class="md-tabs__link">
      Home
    </a>
  </li>

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="./" class="md-tabs__link md-tabs__link--active">
        Day 1
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../day_2/4.bayescan/" class="md-tabs__link">
        Day 2
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Outlier Analysis Workshop" class="md-nav__button md-logo" aria-label="Outlier Analysis Workshop" data-md-component="logo">
      
  <img src="../../theme_images/nesi_ga.png" alt="logo">

    </a>
    Outlier Analysis Workshop
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsAotearoa/Outlier_Analysis_Workshop" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/Outlier_Analysis_Workshop
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
      
      
      
        <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Introduction
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Introduction
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#define-your-working-directory-for-this-project-and-the-vcf-file-location" class="md-nav__link">
    Define your working directory for this project and the VCF file location:
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-data" class="md-nav__link">
    Project data
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../2.pcadapt/" class="md-nav__link">
        PCAdapt
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../3.vcftools/" class="md-nav__link">
        VCFtools
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
        
      
      <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../day_2/4.bayescan/" class="md-nav__link">
        BayeScan
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../day_2/5.baypass/" class="md-nav__link">
        BayPass
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../day_2/6.conclude/" class="md-nav__link">
        Conclusion
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#define-your-working-directory-for-this-project-and-the-vcf-file-location" class="md-nav__link">
    Define your working directory for this project and the VCF file location:
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-data" class="md-nav__link">
    Project data
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="a-brief-introduction-to-genetic-outlier-and-association-analysis">A brief introduction to Genetic Outlier and Association Analysis<a class="headerlink" href="#a-brief-introduction-to-genetic-outlier-and-association-analysis" title="Permanent link">&para;</a></h1>
<p>When we look through a genome to try to find loci that are under divergent selection, two common approaches are outlier analysis and association analysis. <strong>Outlier analysis</strong> requires just knowledge of the genetics of your samples (plus sample metadata, for example, population groupings) and tries to find loci that behave very differently from the underlying patterns across the genome (with the assumption being that the rest of the genome represents patterns of neutral genetic diversity). Meanwhile <strong>association analysis</strong> requires some sort of covariate data and tests whether there are any genetic variants statistically associated with this new data (you may have heard the term <a href="https://www.genome.gov/genetics-glossary/Genome-Wide-Association-Studies">GWAS</a>). This covariate data can come in the form of phenotype data (e.g., morphology, disease status, physiology measures) or could be spatial (e.g., environmental, climate). Association tests look for sites in the genome where the presence or absence of a variant is highly correlated with the values in the co-variate data, usually through some regression-type analysis.</p>
<p>Throughout this vignette, I will collectively refer to outlier and association analysis as selection analysis. There are a lot of programs that exist currently. You can find a very long (though not exhaustive) list <a href="https://bioinformaticshome.com/tools/gwas/gwas.html">here</a>. </p>
<p>This vignette will start by covering some very simple outlier analyses:</p>
<ul>
<li><a href="https://bcm-uga.github.io/pcadapt/articles/pcadapt.html"><strong>PCAdapt</strong></a> can detect genetic marker outliers without having population<sup id="fnref:B"><a class="footnote-ref" href="#fn:B">1</a></sup> designations using a Principle Component Analysis (PCA) approach.</li>
<li><strong>F<sub>ST</sub></strong> outlier analysis is an approach that uses pairwise comparisons between two populations and the fixation index metric to assess each genetic marker.</li>
</ul>
<p>Next, we will conduct some more advanced outlier analyses:</p>
<ul>
<li><a href="https://bcm-uga.github.io/pcadapt/articles/pcadapt.html"><strong>Bayescan</strong></a> looks for differences in allele frequencies between populations to search for outliers.</li>
<li><a href="http://www1.montpellier.inra.fr/CBGP/software/baypass/"><strong>Baypass</strong></a> elaborates on the bayenv model (another popular association analysis program).</li>
</ul>
<p>We will cover the pre-processing of program-specific input files, how to run the programs, how to visualise the output, and in some cases we'll need to take extra steps to map the genetic markers of interest back to the SNP data.</p>
<div class="admonition note">
<p class="admonition-title">Reduced representation versus whole genome sequencing (WGS)</p>
<p>Outlier analysis is often done on reduced representation data. It is important to remember how your genome coverage (the number of genome variant sites / the genome length) will affect your results and interpretation. Often with WGS data, you will see well-resolved 'peaks' with a fairly smooth curve of points leading up to it on either side. From this, we often infer that the highest point is the genetic variant of interest and the sites on either side of the peak exhibit signals of selection because they reside close to, and thus are linked to, the true variant of interest. However, consider that even in WGS data, unless we have every single genetic variant represented (which may not be the case, depending on our variant calling and filtering parameters), it is possible that the genetic variant of interest that we have identified is not the main one, but is simply another neighboring linked SNP to one that is not represented in the data. This problem becomes even more relevant with reduced representation sequencing (RRS), for which the genome coverage may be extremely patchy<sup id="fnref:C"><a class="footnote-ref" href="#fn:C">2</a></sup>. Thus with all outlier analysis, but especially so for those using RRS data, remember that your flagged outliers are not exhaustive and may themselves only be liked to the variant that is truly under selection.</p>
</div>
<figure>
    <a class="glightbox" href="https://els-jbs-prod-cdn.jbs.elsevierhealth.com/cms/asset/4be56b5b-8593-4116-ab9a-ec7b9e3c9a05/gr1.jpg" data-type="image" data-width="100%" data-height="auto" data-desc-position="bottom"><img src="https://els-jbs-prod-cdn.jbs.elsevierhealth.com/cms/asset/4be56b5b-8593-4116-ab9a-ec7b9e3c9a05/gr1.jpg"></a>
    <figcaption>Manhattan plot of the association between P-values and chromosomes.</figcaption>
</figure>

<h2 id="define-your-working-directory-for-this-project-and-the-vcf-file-location">Define your working directory for this project and the VCF file location:<a class="headerlink" href="#define-your-working-directory-for-this-project-and-the-vcf-file-location" title="Permanent link">&para;</a></h2>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>mkdir<span class="w"> </span>~/outlier_analysis
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a><span class="nv">DIR</span><span class="o">=</span>~/outlier_analysis
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a><span class="nb">cd</span><span class="w"> </span><span class="nv">$DIR</span>
</code></pre></div>
</div>
<p>Our data tree will look like this:</p>
<div class="admonition note">
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>outlier_analysis/
<a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>├── analysis
<a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>│   ├── bayescan
<a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a>│   ├── baypass
<a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a>│   ├── pcadapt
<a id="__codelineno-1-6" name="__codelineno-1-6" href="#__codelineno-1-6"></a>│   ├── summary
<a id="__codelineno-1-7" name="__codelineno-1-7" href="#__codelineno-1-7"></a>│   └── vcftools_fst
<a id="__codelineno-1-8" name="__codelineno-1-8" href="#__codelineno-1-8"></a>├── data
<a id="__codelineno-1-9" name="__codelineno-1-9" href="#__codelineno-1-9"></a>├── programs
<a id="__codelineno-1-10" name="__codelineno-1-10" href="#__codelineno-1-10"></a>└── workshop_material
</code></pre></div>
</div>
<p>So let's set up our directories to match this</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>mkdir<span class="w"> </span>-p<span class="w"> </span><span class="o">{</span>analysis/<span class="o">{</span>bayescan,baypass,pcadapt,summary,vcftools_fst<span class="o">}</span>,data,programs,workshop_material<span class="o">}</span>
</code></pre></div>
</div>
<h2 id="project-data">Project data<a class="headerlink" href="#project-data" title="Permanent link">&para;</a></h2>
<p>The data provided in this workshop contains 5007 SNPs loci for 39 individuals (13 individuals each from 3 different locations). This data has some missingness (i.e., missing SNP calls). </p>
<p>There is also a metadata file containing the individual's unique IDs, assigned populations, and a wingspan measurement for each individual. </p>
<p>Let's grab this data from the project's git repository, place the data files into our <code>data</code> directory, and define the environmental variables <code>VCF</code> and <code>METADATA</code> with the locations of the genetic variant and metadata files, respectively.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="c1"># Enter data directory</span>
<a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a><span class="nb">cd</span><span class="w"> </span><span class="nv">$DIR</span>/data
<a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>
<a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a><span class="c1"># Download required files to data directory</span>
<a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a>wget<span class="w"> </span>https://raw.githubusercontent.com/GenomicsAotearoa/Outlier_Analysis_Workshop/main/workshop_files/starling_3populations.recode.vcf
<a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a>wget<span class="w"> </span>https://raw.githubusercontent.com/GenomicsAotearoa/Outlier_Analysis_Workshop/main/workshop_files/starling_3populations_metadata.txt
<a id="__codelineno-3-7" name="__codelineno-3-7" href="#__codelineno-3-7"></a>
<a id="__codelineno-3-8" name="__codelineno-3-8" href="#__codelineno-3-8"></a><span class="c1"># Set environment variables</span>
<a id="__codelineno-3-9" name="__codelineno-3-9" href="#__codelineno-3-9"></a><span class="nv">VCF</span><span class="o">=</span><span class="nv">$DIR</span>/data/starling_3populations.recode.vcf
<a id="__codelineno-3-10" name="__codelineno-3-10" href="#__codelineno-3-10"></a><span class="nv">METADATA</span><span class="o">=</span><span class="nv">$DIR</span>/data/starling_3populations_metadata.txt
</code></pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Working with your own data</p>
<p>Alternatively, you can also use your own data for this workshop. If so, it is a good idea to thin your SNP dataset down to roughly 5,000 SNPs to ensure compute times are not too long. If you have more than 50 individuals, you may also want to reduce this. If you would like to do this, place your genetic variant and metadata file in the <code>data</code> directory and define <code>VCF</code> and <code>METADATA</code> based on their names.</p>
</div>
<p>Across this workshop, we will need the genetic data to be in several different formats. Let's prepare that now. First, we convert the VCF to PLINK and then to BED.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="c1"># Enter data directory</span>
<a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a><span class="nb">cd</span><span class="w"> </span><span class="nv">$DIR</span>/data
<a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>
<a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a><span class="c1"># Load modules</span>
<a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a>module<span class="w"> </span>purge
<a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a>module<span class="w"> </span>load<span class="w"> </span>VCFtools/0.1.15-GCC-9.2.0-Perl-5.30.1
<a id="__codelineno-4-7" name="__codelineno-4-7" href="#__codelineno-4-7"></a>module<span class="w"> </span>load<span class="w"> </span>PLINK/1.09b6.16<span class="w"> </span>
<a id="__codelineno-4-8" name="__codelineno-4-8" href="#__codelineno-4-8"></a>
<a id="__codelineno-4-9" name="__codelineno-4-9" href="#__codelineno-4-9"></a><span class="c1"># Convert VCF to PLINK</span>
<a id="__codelineno-4-10" name="__codelineno-4-10" href="#__codelineno-4-10"></a>vcftools<span class="w"> </span>--vcf<span class="w"> </span><span class="nv">$VCF</span><span class="w"> </span>--plink<span class="w"> </span>--out<span class="w"> </span>starling_3populations.plink
<a id="__codelineno-4-11" name="__codelineno-4-11" href="#__codelineno-4-11"></a>
<a id="__codelineno-4-12" name="__codelineno-4-12" href="#__codelineno-4-12"></a><span class="c1"># Convert PLINK to BED</span>
<a id="__codelineno-4-13" name="__codelineno-4-13" href="#__codelineno-4-13"></a>plink<span class="w"> </span>--file<span class="w"> </span>starling_3populations.plink<span class="w"> </span>--make-bed<span class="w"> </span>--noweb<span class="w"> </span>--out<span class="w"> </span>starling_3populations
</code></pre></div>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>VCFtools - 0.1.15
<a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a>(C) Adam Auton and Anthony Marcketta 2009
<a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a>
<a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a>Parameters as interpreted:
<a id="__codelineno-5-5" name="__codelineno-5-5" href="#__codelineno-5-5"></a>        --vcf /nesi/nobackup/nesi02659/outlier_analysis_workshop/data/starling_3populations.recode.vcf
<a id="__codelineno-5-6" name="__codelineno-5-6" href="#__codelineno-5-6"></a>        --out starling_3populations.plink
<a id="__codelineno-5-7" name="__codelineno-5-7" href="#__codelineno-5-7"></a>        --plink
<a id="__codelineno-5-8" name="__codelineno-5-8" href="#__codelineno-5-8"></a>
<a id="__codelineno-5-9" name="__codelineno-5-9" href="#__codelineno-5-9"></a>After filtering, kept 39 out of 39 Individuals
<a id="__codelineno-5-10" name="__codelineno-5-10" href="#__codelineno-5-10"></a>Writing PLINK PED and MAP files ... 
<a id="__codelineno-5-11" name="__codelineno-5-11" href="#__codelineno-5-11"></a>
<a id="__codelineno-5-12" name="__codelineno-5-12" href="#__codelineno-5-12"></a>Unrecognized values used for CHROM: starling4 - Replacing with 0.
<a id="__codelineno-5-13" name="__codelineno-5-13" href="#__codelineno-5-13"></a>Done.
<a id="__codelineno-5-14" name="__codelineno-5-14" href="#__codelineno-5-14"></a>After filtering, kept 5007 out of a possible 5007 Sites
<a id="__codelineno-5-15" name="__codelineno-5-15" href="#__codelineno-5-15"></a>Run Time = 0.00 seconds
</code></pre></div>
</details>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>PLINK v1.90b6.16 64-bit (19 Feb 2020)          www.cog-genomics.org/plink/1.9/
<a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3
<a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>Logging to starling_3populations.log.
<a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a>Options in effect:
<a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a>  --file starling_3populations.plink
<a id="__codelineno-6-6" name="__codelineno-6-6" href="#__codelineno-6-6"></a>  --make-bed
<a id="__codelineno-6-7" name="__codelineno-6-7" href="#__codelineno-6-7"></a>  --noweb
<a id="__codelineno-6-8" name="__codelineno-6-8" href="#__codelineno-6-8"></a>  --out starling_3populations
<a id="__codelineno-6-9" name="__codelineno-6-9" href="#__codelineno-6-9"></a>
<a id="__codelineno-6-10" name="__codelineno-6-10" href="#__codelineno-6-10"></a>Note: --noweb has no effect since no web check is implemented yet.
<a id="__codelineno-6-11" name="__codelineno-6-11" href="#__codelineno-6-11"></a>128824 MB RAM detected; reserving 64412 MB for main workspace.
<a id="__codelineno-6-12" name="__codelineno-6-12" href="#__codelineno-6-12"></a>.ped scan complete (for binary autoconversion).
<a id="__codelineno-6-13" name="__codelineno-6-13" href="#__codelineno-6-13"></a>Performing single-pass .bed write (5007 variants, 39 people).
<a id="__codelineno-6-14" name="__codelineno-6-14" href="#__codelineno-6-14"></a>--file: starling_3populations-temporary.bed +
<a id="__codelineno-6-15" name="__codelineno-6-15" href="#__codelineno-6-15"></a>starling_3populations-temporary.bim + starling_3populations-temporary.fam
<a id="__codelineno-6-16" name="__codelineno-6-16" href="#__codelineno-6-16"></a>written.
<a id="__codelineno-6-17" name="__codelineno-6-17" href="#__codelineno-6-17"></a>5007 variants loaded from .bim file.
<a id="__codelineno-6-18" name="__codelineno-6-18" href="#__codelineno-6-18"></a>39 people (0 males, 0 females, 39 ambiguous) loaded from .fam.
<a id="__codelineno-6-19" name="__codelineno-6-19" href="#__codelineno-6-19"></a>Ambiguous sex IDs written to starling_3populations.nosex .
<a id="__codelineno-6-20" name="__codelineno-6-20" href="#__codelineno-6-20"></a>Using 1 thread (no multithreaded calculations invoked).
<a id="__codelineno-6-21" name="__codelineno-6-21" href="#__codelineno-6-21"></a>Before main variant filters, 39 founders and 0 nonfounders present.
<a id="__codelineno-6-22" name="__codelineno-6-22" href="#__codelineno-6-22"></a>Calculating allele frequencies... done.
<a id="__codelineno-6-23" name="__codelineno-6-23" href="#__codelineno-6-23"></a>Total genotyping rate is 0.723331.
<a id="__codelineno-6-24" name="__codelineno-6-24" href="#__codelineno-6-24"></a>5007 variants and 39 people pass filters and QC.
<a id="__codelineno-6-25" name="__codelineno-6-25" href="#__codelineno-6-25"></a>Note: No phenotypes present.
<a id="__codelineno-6-26" name="__codelineno-6-26" href="#__codelineno-6-26"></a>--make-bed to starling_3populations.bed + starling_3populations.bim +
<a id="__codelineno-6-27" name="__codelineno-6-27" href="#__codelineno-6-27"></a>starling_3populations.fam ... done.
</code></pre></div>
</details>
</div>
<div class="footnote">
<hr />
<ol>
<li id="fn:B">
<p>I will say population for simplicity throughout this vignette. However, equally we can test for differences between sample sites, subpopulations, and other types of groupings. What counts as one 'group' of organisms will depend on your study system or question.&#160;<a class="footnote-backref" href="#fnref:B" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="fn:C">
<p>You may also want to consider linkage blocks.&#160;<a class="footnote-backref" href="#fnref:C" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
</ol>
</div>





                
              </article>
            </div>
          
          
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky"], "search": "../../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.b4d07000.min.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  <script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>